cms/tests/test_views.py

```diff
@@ -0,0 +1,70 @@
+from django.core.urlresolvers import reverse
+from django.contrib.auth.models import User
+from django.test.client import Client
+
+from cms.tests.base import BaseCmsTestCase
+
+from unicore.content import models as eg_models
+
+
+class TestCustomAdminViews(BaseCmsTestCase):
+
+    def setUp(self):
+        self.user = User.objects.create_superuser(
+            'testuser', 'test@email.com', password='test')
+        self.client = Client()
+        self.client.login(username='testuser', password='test')
+
+        self.workspace = self.mk_workspace()
+        self.remote_workspace = self.mk_workspace(
+            name='%s_remote' % (self.workspace.index_prefix,),
+            index_prefix='%s_remote' % (self.workspace.index_prefix,))
+
+        self.create_localisation(self.workspace)
+        self.create_categories(self.workspace)
+        self.create_pages(self.workspace)
+
+        # create content for source repo
+        self.create_localisation(self.remote_workspace, locale='spa_ES')
+        self.create_categories(self.remote_workspace, locale='spa_ES')
+        self.create_pages(self.remote_workspace, locale='spa_ES')
+
+        self.create_localisation(self.remote_workspace, locale='fre_FR')
+        self.create_categories(self.remote_workspace, locale='fre_FR')
+        self.create_pages(self.remote_workspace, locale='fre_FR')
+
+    def test_import_repo(self):
+        self.workspace.refresh_index()
+        self.remote_workspace.refresh_index()
+
+        with self.settings(IMPORT_CLONE_REPO_PATH='.test_repos/',
```

Comment:
@miltontony I mean that if you use `with self.settings(ELASTICSEARCH_HOST='new_host')`, the default value of es in the `setup_workspace` function won't be `{'urls': ['new_host']}` - it will have the original setting value.

Not a major issue. Just unexpected.